llvm/lib/CodeGen/IntrinsicLowering.cpp

```diff
@@ -199,6 +199,25 @@ static Value *LowerCTLZ(LLVMContext &Context, Value *V, Instruction *IP) {
   return LowerCTPOP(Context, V, IP);
 }
 
+/// Emit the code to lower clmul of V1, V2 before the specified instruction IP.
+static Value *LowerCLMUL(LLVMContext &Context, Value *V1, Value *V2, Instruction *IP) {
+
+  IRBuilder<> Builder(IP);
+
+  unsigned BitSize = V1->getType()->getPrimitiveSizeInBits();
```

Comment:
I'm assuming this doesn't work correctly with vectors?

Reply:
it looks like there's `getScalarSizeInBits`. Do you happen to know if this works for both primitives and vectors?

Reply:
I think it should, although generally shouldn't use getPrimitiveSizeInBits (but it's probably only an issue with pointers, which don't apply here)